src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -0,0 +1,467 @@
+    // https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/3724
+    // behavior is not specified when Abs * values are not present and user values are present
+    // implemented behavior accepts the user values without regard to default Abs values.
+
+    // Per global matter data model policy
+    // if a attribute is not present then it's default shall be used.
+
+    status = GetAbsMinHeatSetpointLimit(endpoint, &AbsMinHeatSetpointLimit);
+    status = GetAbsMaxHeatSetpointLimit(endpoint, &AbsMaxHeatSetpointLimit);
+
+    status = GetMinHeatSetpointLimit(endpoint, &MinHeatSetpointLimit);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        MinHeatSetpointLimit = AbsMinHeatSetpointLimit;
+    }
+
+    status = GetMaxHeatSetpointLimit(endpoint, &MaxHeatSetpointLimit);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        MaxHeatSetpointLimit = AbsMaxHeatSetpointLimit;
+    }
+
+    // Make sure the user imposed limits are within the manufacturer imposed limits
+
+    // https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/3725
+    // Spec does not specify the behavior is the requested setpoint exceeds the limit allowed
+    // This implementation clamps at the limit.
+
+    // resolution of 3725 is to clamp.
+
+    if (MinHeatSetpointLimit < AbsMinHeatSetpointLimit)
+        MinHeatSetpointLimit = AbsMinHeatSetpointLimit;
+
+    if (MaxHeatSetpointLimit > AbsMaxHeatSetpointLimit)
+        MaxHeatSetpointLimit = AbsMaxHeatSetpointLimit;
+
+    if (HeatingSetpoint < MinHeatSetpointLimit)
+        HeatingSetpoint = MinHeatSetpointLimit;
+
+    if (HeatingSetpoint > MaxHeatSetpointLimit)
+        HeatingSetpoint = MaxHeatSetpointLimit;
+
+    return HeatingSetpoint;
+}
+
+int16_t EnforceCoolingSetpointLimits(int16_t CoolingSetpoint, EndpointId endpoint)
+{
+    // Optional Mfg supplied limits
+    int16_t AbsMinCoolSetpointLimit = 1600; // 16C (61 F) is the default
+    int16_t AbsMaxCoolSetpointLimit = 3200; // 32C (90 F) is the default
+
+    // Optional User supplied limits
+    int16_t MinCoolSetpointLimit = 1600; // 16C (61 F) is the default
+    int16_t MaxCoolSetpointLimit = 3200; // 32C (90 F) is the default
+
+    // Attempt to read the setpoint limits
+    // Absmin/max are manufacturer limits
+    // min/max are user imposed min/max
+
+    // Note that the limits are initialized above per the spec limits
+    // if they are not present emberAfReadAttribute() will not update the value so the defaults are used
+    EmberAfStatus status;
+
+    // https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/3724
+    // behavior is not specified when Abs * values are not present and user values are present
+    // implemented behavior accepts the user values without regard to default Abs values.
+
+    // Per global matter data model policy
+    // if a attribute is not present then it's default shall be used.
+
+    status = GetAbsMinCoolSetpointLimit(endpoint, &AbsMinCoolSetpointLimit);
```

Comment:
same here: status checking and logging seems useful

Reply:
Warning: "AbsMaxHeatSetpointLimit missing using default"